tests/hw/esp32/CMakeLists.txt

```diff
@@ -96,11 +96,36 @@ function(parse_args_from_file file_path result_var)
   set(${result_var} ${result_list} PARENT_SCOPE)
 endfunction()
 
+function(parse_wireless_from_file file_path result_var)
```

Comment:
It is a bit weird that you map from WIRELESS to flaky. Maybe make the category hw-wireless? You do more than just retries in the sense that you also separate them from the other tests.